queue_job/controllers/main.py

```diff
@@ -71,10 +71,16 @@ def runjob(self, db, job_uuid, **kw):
         env = http.request.env(user=odoo.SUPERUSER_ID)
 
         def retry_postpone(job, message, seconds=None):
-            job.postpone(result=message, seconds=seconds)
-            job.set_pending(reset_retry=False)
-            job.store()
-            env.cr.commit()
+
+            job.env.cr.rollback()
```

Comment:
shouldn't you rollback only in case of errors? /cc @guewen

Reply:
I think it's there to prevent a conflict between the job.env's transaction and the new one. The job record might have been updated by the first one and will be updated by the second one? It's not called in the FailedJobError case though so why is it necessary here?